File: internal/models/fixed_cost.go
package models

type FixedCost struct {
	ID        int    `json:"id"`
	UserID    string `json:"user_id"`
	Name      string `json:"name"`
	Amount    int    `json:"amount"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

File: frontend/src/lib/types/fixed-cost.ts
export type FixedCost = {
  id: number
  user_id: string
  name: string
  amount: number
  created_at: string
  updated_at: string
}

export type CreateFixedCostInput = {
  name: string
  amount: number
}

export type UpdateFixedCostInput = {
  name: string
  amount: number
}

export type CreateFixedCostResponse = {
  fixed_cost: FixedCost
}

export type GetFixedCostsResponse = {
  fixed_costs: FixedCost[]
}

export type UpdateFixedCostResponse = {
  fixed_cost: FixedCost
}
